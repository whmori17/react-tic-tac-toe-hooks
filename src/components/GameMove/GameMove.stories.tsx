import * as React from 'react';
import { GameMove } from '.';

export default {
  title: 'GameMove',
};

const onClick = (i: number) => console.log(i);

export const gameStartMove = () => <GameMove step={0} {...{ onClick }} />;
export const stepOneGameMove = () => <GameMove step={1} {...{ onClick }} />;
