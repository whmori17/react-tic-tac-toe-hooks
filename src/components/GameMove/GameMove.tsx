import React from 'react';

export interface GameMoveProps {
  step: number;
  onClick: (move: number) => void;
}

export function GameMove(props: GameMoveProps) {
  const desc = props.step ? 'Go to move #' + props.step : 'Go to game start';
  return (
    <li key={props.step}>
      <button onClick={() => props.onClick(props.step)}>{desc}</button>
    </li>
  );
}
