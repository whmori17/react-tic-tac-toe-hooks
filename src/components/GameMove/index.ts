export { GameMove, GameMoveProps } from './GameMove';
