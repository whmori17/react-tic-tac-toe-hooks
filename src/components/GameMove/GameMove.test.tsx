import * as React from 'react';
import { shallow } from 'enzyme';
import { GameMove, GameMoveProps } from './GameMove';

describe('GameMove', () => {
  const gameMoveProps: GameMoveProps = {
    onClick: (move: number) => console.log(move),
    step: 1,
  };

  it('should render', () => {
    const wrapper = shallow(<GameMove {...gameMoveProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with step desc', () => {
    const wrapper = shallow(<GameMove {...gameMoveProps} />);

    expect(wrapper.find('button').contains('Go to move #1')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with game start desc', () => {
    const gameMoveStartProps: GameMoveProps = {
      onClick: (move: number) => console.log(move),
      step: 0,
    };
    const wrapper = shallow(<GameMove {...gameMoveStartProps} />);

    expect(wrapper.find('button').contains('Go to game start')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger onClick when clicked', () => {
    const mockCallBack = jest.fn();
    const gameMoveStartProps: GameMoveProps = {
      onClick: mockCallBack,
      step: 0,
    };
    const wrapper = shallow(<GameMove {...gameMoveStartProps} />);

    wrapper.find('button').simulate('click');

    expect(mockCallBack.mock.calls.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
