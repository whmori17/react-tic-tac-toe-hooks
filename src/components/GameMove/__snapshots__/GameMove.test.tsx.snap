// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMove should render 1`] = `
<li
  key="1"
>
  <button
    onClick={[Function]}
  >
    Go to move #1
  </button>
</li>
`;

exports[`GameMove should render with game start desc 1`] = `
<li
  key="0"
>
  <button
    onClick={[Function]}
  >
    Go to game start
  </button>
</li>
`;

exports[`GameMove should render with step desc 1`] = `
<li
  key="1"
>
  <button
    onClick={[Function]}
  >
    Go to move #1
  </button>
</li>
`;

exports[`GameMove should trigger onClick when clicked 1`] = `
<li
  key="0"
>
  <button
    onClick={[Function]}
  >
    Go to game start
  </button>
</li>
`;
